ode-mongo-logger/src/main/java/com/trihydro/mongologger/app/loggers/MongoLogger.java

```diff
@@ -22,26 +23,30 @@
 public class MongoLogger {
 
     private String serverAddress;
-    private String username;
-    private String password;
     private String databaseName;
-    private String authDatabaseName;
     private MongoCredential credential;
     private Utility utility;
     private EmailHelper emailHelper;
     private MongoLoggerConfiguration config;
+    private MongoClient _mongoClient;
 
     @Autowired
     public void InjectDependencies(MongoLoggerConfiguration _config, Utility _utility, EmailHelper _emailHelper) {
         config = _config;
-        username = config.getMongoUsername(); // the user name
+        String username = config.getMongoUsername(); // the user name
         databaseName = config.getMongoDatabase(); // the name of the database to deposit records into
-        authDatabaseName = config.getMongoAuthDatabase(); // the name of the database in which the user is defined
-        password = config.getMongoPassword(); // the password as a character array
+        String authDatabaseName = config.getMongoAuthDatabase(); // the name of the database in which the user is defined
+        String password = config.getMongoPassword(); // the password as a character array
         serverAddress = config.getMongoHost();
         credential = MongoCredential.createCredential(username, authDatabaseName, password.toCharArray());
         utility = _utility;
         emailHelper = _emailHelper;
+        configureMongoClient();
+    }
+
+    private void configureMongoClient() {
+        _mongoClient =
+            MongoClients.create(MongoClientSettings.builder().applyToClusterSettings(builder -> builder.hosts(List.of(new ServerAddress(serverAddress, 27017)))).credential(credential).build());
     }

```

Comment:
suggestion(non-blocking): make this more readable (and debuggable because more linebreaks means more descriptive error messages)

```suggestion
    private MongoClient configureMongoClient() {
        var hosts = List.of(new ServerAddress(serverAddress, 27017));
        var settings = MongoClientSettings.builder()
                .applyToClusterSettings(builder -> builder.hosts(hosts)).credential(credential)
                .build();
        return MongoClients.create(settings);
    }
```

Reply:
Committed your suggestion!